json-schemas/interface-methods/protocol-rule-set.json

```
@@ -84,6 +84,10 @@
     "$contextRole": {
       "$comment": "When `true`, this turns a record into `role` that may be used within a context",
       "type": "boolean"
+    },
+    "$sizeLimit": {
+      "$comment": "When set, this limits the size of the record in bytes",
+      "type": "number"
```

Suggestion: implement a minimum of 0 or 1 so that schema validation blows up if they put a negative number.

@thehenrytsai @csuwildcat what are your thoughts of allowing a zero to represent no data any usefulness to that? Otherwise we can do a minimum of 1 byte?

```suggestion
      "type": "number",
      "minimum": 0
```